modules/instrument_manager/php/instrument_manager.class.inc

```diff
@@ -168,24 +155,38 @@ class Instrument_Manager extends \NDB_Menu_Filter
              * installation is disabled (ie. has no quatUser), as the above
              * error will return before this one.
             */
-            return (new \Loris\Http\Response())
+            return (new \LORIS\Http\Response())
                 ->withStatus(409, 'Conflict')
                 ->withBody(
                     new \LORIS\Http\StringStream(
                         json_encode(
-                            array('error' => self::FILE_ALREADY_EXISTS)
+                            array('error' => self::ALREADY_INSTALLED)
                         )
                     )
                 );
         }
 
-        $uploaded_file->moveTo($fullpath);
-        chmod($fullpath, 0644);
+        /* Although only .linst files should be uploaded to this module, and
+         * they are just special text files, they are marked as
+         * application/octet-stream upon upload.
+         */
+
+        $uploader = (new \LORIS\FilesUploadHandler($targetdir))
+            ->withPermittedMIMETypes(
+                'application/octet-stream'
+            );
+
+        $response = $uploader->handle($request);
+
+        if ($response->getStatusCode() !== 200) {
+            return $response;
+        }
 
         // Scripts in tools/ often make relative imports, so we must change
         // our effective directory in order to use them.
         chdir($this->path . "/tools");
         // Use tools/ script to generate an SQL patch file based on the
+        //
```

Comment:
This line is not needed

Reply:
@johnsaigle please stop marking other people's comments as "Resolved" when they haven't been resolved.

Reply:
this line is still not needed